File: backend/src/main/java/com/magesty/backend/models/SocialMedia.java
package com.magesty.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.magesty.backend.models.dto.SocialMediaDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;
import java.util.UUID;

import static javax.persistence.FetchType.*;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
public class SocialMedia implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String mediaName;
    private String socialUrl;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "media_id")
    private Admin admin;

    public static SocialMedia from(SocialMediaDto socialMediaDto){
        SocialMedia socialMedia = new SocialMedia();

        if(Objects.isNull(socialMediaDto)){
            return null;
        }else {
            socialMedia.setSocialUrl(socialMediaDto.getSocialUrl());
            socialMedia.setMediaName(socialMediaDto.getMediaName());
            socialMedia.setId(socialMediaDto.getId());

            return socialMedia;
        }
    }

}

File: frontend-ng/src/app/models/SocialMedia/social-media.ts
export class SocialMedia {
  private _id: number;
  private _mediaName: string;
  private _socialUrl: string;

  constructor(id: number, mediaName: string, socialUrl: string) {
    this._id = id;
    this._mediaName = mediaName;
    this._socialUrl = socialUrl;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get mediaName(): string {
    return this._mediaName;
  }

  set mediaName(value: string) {
    this._mediaName = value;
  }

  get socialUrl(): string {
    return this._socialUrl;
  }

  set socialUrl(value: string) {
    this._socialUrl = value;
  }
}
